File: flutter-sdk/lib/src/domain/models/chat_event/chat_event.dart
// Copyright (c) Yalochat, Inc. All rights reserved.

import 'package:equatable/equatable.dart';

sealed class ChatEvent {}

final class TypingStart extends ChatEvent with EquatableMixin {
  final String statusText;

  TypingStart({required this.statusText});

  @override
  List<Object?> get props => [statusText];
}

final class TypingStop extends ChatEvent with EquatableMixin {
  @override
  List<Object?> get props => [];
}

File: typescript-sdk/packages/core/src/domain/chat-event.ts
// Copyright (c) Yalochat, Inc. All rights reserved.

export type ChatEvent =
  | { type: 'typingStart'; statusText: string }
  | { type: 'typingStop' };

export function typingStart(statusText: string): ChatEvent {
  return { type: 'typingStart', statusText };
}

export function typingStop(): ChatEvent {
  return { type: 'typingStop' };
}
